spec.bs

```diff
@@ -980,6 +980,77 @@ To <dfn>add a part</dfn> given a [=pattern parser=] |parser|, a string |prefix|,
 
 <h3 id=converting-part-lists-to-regular-expressions>Converting Part Lists to Regular Expressions</h3>
 
+<div algorithm>
+To <dfn export>generate a regular expression and name list</dfn> from a given [=/part list=] |part list| and [=/options=] |options|:
+
+  1. Let |result| be "`^`".
+  1. Let |name list| be a new [=list=].
+  1. [=list/For each=] |part| of |part list|:
+    1. If |part|'s [=part/type=] is "<a for=part/type>`fixed-text`</a>":
+      1. If |part|'s [=part/modifier=] is "<a for=part/modifier>`none`</a>", then append the result of running [=escape a regexp string=] given |part|'s [=part/value=] to the end of |result|.
+      1. Else:
+        <div class=note>
+        <p>A "<a for=part/type>`fixed-text`</a>" |part| with a modifier uses a non capturing group with the form:
+        <p>`(?:<fixed text>)<modifier>`
```

Comment:
```suggestion
        <p>`(?:<fixed text>)<modifier>`.
```

Reply:
As discussed in chat, don't want to put a period at the end here since it might get confused as a regexp char.  I'll refactor to avoid this.